Sources/Networking/HTTPClient/ETagManager.swift

```diff
@@ -185,11 +185,12 @@ extension ETagManager.Response {
         return try? JSONEncoder.default.encode(self)
     }
 
-    fileprivate var asResponse: HTTPResponse<Data> {
+    fileprivate func asResponse(withRequestDate requestDate: Date?) -> HTTPResponse<Data> {
         return HTTPResponse(
             statusCode: self.statusCode,
             responseHeaders: [:],
             body: self.data,
+            requestDate: requestDate,
             verificationResult: self.verificationResult
```

Comment:
We should only update the requestDate if the verification result of the NOT_MODIFIED response is green, right?

Reply:
And do we persist this date change in the cache?
Imagine:
1) 200 OK, valid signature, with request time t=1
2) 304 NOT_MODIFIED, valid signature, with request time t=2
3) Timeout or error 500... What the SDK will retrieve t=1 or t=2. Should be t=2 (and valid signature)

Reply:
Similarly:
1) 200 OK, valid signature with request time t=1. SDK gets (valid signature, t=1)
2) 304 NOT_MODIFIED, INVALID signature, with request time t=2. SDK should get (valid signature, t=1) 
3) Timeout or error 500, SDK should get (valid signature, t=1) 

1) 200 OK valid signature, t=1. SDK gets (valid signature, t=1)
2) 200 OK INVALID SIGNATURE, t=2. SDK gets (valid signature, t=1)
3) Timeout or error 500, SDK should get (valid signature, t=1)

Reply:
Nacho can correct me but I believe we currently cache the request time not at the response level but at the model level. Right now we only cache it for the CustomerInfo model so we only store it in those models. We don't update the response cache on not modified responses either

Reply:
Short answer is t2.

@bisho you're conflating 2 separate layers of caching (which is normal because you're not familiar with the codebase).

If the server fails to return a 304 (500 or timeout), `HTTPClient` won't return the cached data.
That caching would be handled by `DeviceCache`, which would receive the updated request time.

Reply:
> We should only update the requestDate if the verification result of the NOT_MODIFIED response is green, right?

That I think I have a test for but let me check.

Reply:
> 304 NOT_MODIFIED, INVALID signature, with request time t=2. SDK should get (valid signature, t=1)

Oh that's a case I didn't consider.
Are we sure we want to do that? That would break an app without any indication that this is happening.

Reply:
> We should only update the requestDate if the verification result of the NOT_MODIFIED response is green, right?

That's already done:
```swift
fileprivate func copyWithNewRequestDate() -> Self {
    // Update request time from server unless it failed verification.
    guard self.verificationResult != .failed, let requestDate = self.requestDate else { return self }
```

Reply:
>> 304 NOT_MODIFIED, INVALID signature, with request time t=2. SDK should get (valid signature, t=1)

> Oh that's a case I didn't consider.
> Are we sure we want to do that? That would break an app without any indication that this is happening.

The thing is that you can't use t=2 as it is not verified. And making the cache not verified in whole defeats the purpose of the cache. Note that there are legit cases of tampering (you are in an airport wifi that intercepts HTTPs. So you can't assume it is a bad actor and use t=2 but making it fail validation. We have a cache, use it until you can. The only case for using t=2 is if there app has disabled verification.

Reply:
I agree that we can't use t=2, otherwise that opens us up to an easy attack. 

But I'm not sure about the part about considering any result that we have "verified" if we get a signature that we can't verify - even if it's from a 304. Like, "_someone_ told us to use what's in cache but we don't really trust them" seems like a clear use case for "not verified". 

I get that we're not entirely sure if any valid tampering will happen in real life, but... this is all happening at the application level, right? Like, there really shouldn't be any modified content in the response content, datetime or request uuid that a public wifi modifies to have us fail validation. 

It feels like we might be setting ourselves up for one of those use cases that takes a long time to explain to someone, while just saying "if you got an invalid signature, you get an unverified entitlement" seems a lot simpler